docs/topics/testing/tools.txt

```diff
@@ -1755,6 +1755,60 @@ You can also exclude tests by tag. To run core tests if they are not slow:
 test has two tags and you select one of them and exclude the other, the test
 won't be run.
 
+.. _async-tests:
+
+Testing Asynchronous Code
+=========================
+
+.. versionadded:: 3.1
+
+If you merely want to test the output of your asynchronous views, the standard
+test client will run them inside their own asynchronous loop without any extra
+work needed on your part.
+
+However, if you want to write fully-asynchronous tests for a Django project, you
+will need to take several things into account.
+
+Firstly, your tests must be ``async def`` methods on the test class (in order
+to give them an asynchronous context). Django will automatically detect
+any ``async def`` tests and wrap them so they run in their own event loop.
+
+If you are testing from an asynchronous function, you must also use the
+asynchronous test client. This is available as ``django.test.AsyncClient``,
+or as ``self.async_client`` on any test.
+
+It has the same methods and signatures as the synchronous (normal) test client,
```

Comment:
This is not entirely true, we should at least add a note that HTTP methods: `get()`, `post()`, `head()`, `options()`, `put()`, `patch()`, and `trace()` doesn't support the `follow` argument.

Reply:
I adjusted `AsyncClient` to raise `NotImplemented` is you pass the `follow` parameter, plus tests in d678b36c1c831b5b54b007cc1bf47be644b8e5a2.